src/components/VideoControls/VideoControls.js

```diff
@@ -0,0 +1,149 @@
+import React, { useEffect, useState, useRef } from "react";
+import playIcon from "../../play.svg";
+import speakerIcon from "../../volume-up.svg";
+import fullscreen from "../../expand.svg";
+import disableFullscreen from "../../compress.svg";
+import pauseIcon from "../../pause.svg";
+import mutedIcon from "../../volume-mute.svg";
+import { secondsToTime } from "../CustomVideoControls/time";
+
+const VideoControls = (props) => {
+  const { video, videoContainer } = props;
+  const [pausePlayIcon, setPauseplayIcon] = useState(pauseIcon);
+  const [muteUnmuteIcon, setMuteUnmuteIcon] = useState(speakerIcon);
+  const [
+    enableDisableFullscreenIcon,
+    setEnableDisableFullscreenIcon,
+  ] = useState(fullscreen);
+  const [totalDuration, setTotalDuration] = useState(secondsToTime(0));
+  const [realTime, setRealTime] = useState(secondsToTime(0));
+
+  const progress = useRef(null);
+  const volumeSlider = useRef(null);
+  const seek = useRef(null);
+  const controls = useRef(null);
+
+  useEffect(() => {
+    video.addEventListener("loadedmetadata", () => {
+      setTotalDuration(secondsToTime(video.duration));
+    });
+    video.addEventListener("timeupdate", () => {
+      setRealTime(secondsToTime(video.currentTime));
+      const progressWidth = (video.currentTime / video.duration) * 100;
+      progress.current.style.width = `${progressWidth}%`;
+    });
+    video.addEventListener("ended", () => {
+      setPauseplayIcon(playIcon);
+    });
+    video.addEventListener("click", () => {
+      playVideo();
+    });
+    videoContainer.addEventListener("mouseover", () => {
+      showControls();
+    });
+    videoContainer.addEventListener("mouseout", () => {
+      hideControls();
+    });
+  }, [video]);
+
+  const showControls = () => {
+    controls.current.classList.remove("hide");
+  };
+
+  const hideControls = () => {
+    controls.current.classList.add("hide");
+  };
+
+  const playVideo = () => {
+    if (video.paused) {
+      video.play();
+    } else {
+      video.pause();
+    }
+    setPauseplayIcon(video.paused ? playIcon : pauseIcon);
+  };
+
+  const muteVideo = () => {
+    if (video.muted) {
+      video.muted = false;
+    } else {
+      video.muted = true;
+    }
```

Comment:
we can change this block to:

```suggestion
    video.muted = !video.muted;
```

Reply:
Noted. Will do